docs/Testing.md

```diff
@@ -0,0 +1,99 @@
+Testing
+=======
+
+This document details how to test different components of the SNMP Alarm Agent. 
+
+Unit Testing
+------------
+
+To run the SNMP Alarm Agent unit test suite, change to the top-level
+clearwater-snmp-handlers directory and issue make test (this will also run the
+integration tests outlined below).
+
+SNMP Alarm Agent unit tests use the Google Test framework and are responsible
+for testing the following components of the SNMP Alarm Agent:
+
+- The JSON Parser
+  - Tests that any errors in the JSON alarm definitions files (for example the
+    alarm description being too long) will cause the tables to not initialise
+    with an appropriate log error message.
+- The alarm req listener
+  - Tests that appropriate traps are being sent whenever alarms are cleared or
+    raised.
+
+Note that to ensure that we do not call any SNMP commands in unit tests we
+include fake SNMP functions with no useful output. These functions are used in
+place of real SNMP commands  whenever make test is issued.
+
+The output from the unit test run will look like this:
+```
+[==========] Running 31 tests from 3 test cases.
+[----------] Global test environment set-up.
+[----------] 12 tests from AlarmTableDefsTest
+[ RUN      ] AlarmTableDefsTest.InitializationNoFiles
+[       OK ] AlarmTableDefsTest.InitializationNoFiles (0 ms)
+[ RUN      ] AlarmTableDefsTest.InitializationMultiDef
+[       OK ] AlarmTableDefsTest.InitializationMultiDef (0 ms)
+.
+.
+.
+.
+[ RUN      ] AlarmReqListenerZmqErrorTest.CloseSocket
+[       OK ] AlarmReqListenerZmqErrorTest.CloseSocket (0 ms)
+[ RUN      ] AlarmReqListenerZmqErrorTest.DestroyContext
+[       OK ] AlarmReqListenerZmqErrorTest.DestroyContext (0 ms)
+[----------] 9 tests from AlarmReqListenerZmqErrorTest (2 ms total)
+
+[----------] Global test environment tear-down
+[==========] 31 tests from 3 test cases ran. (316 ms total)
+[  PASSED  ] 31 tests.
+```
+
+Integration Testing
+-------------------
+
+To run the SNMP Alarm Agent integration test suite, change to the top-level
+clearwater-snmp-handlers director and issue make fvtest.
+
+The integration tests also use the Google Test framework and are responsible for
+testing the following components of the SNMP Alarm Agent.
+
+- The Alarm Model Table
+- The itu Alarm Table
+  - Tests that when quieried using snmp get commands these tables will output the
```

Comment:
Also I'd say either "snmpget" (the command-line command with no space) or "SNMP GET" (the protocol and the PDU, both of which are capitalised).